src/hawkbit-client.c

```diff
@@ -256,6 +274,24 @@ static void set_default_curl_opts(CURL *curl)
         curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, hawkbit_config->connect_timeout);
         curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, hawkbit_config->ssl_verify ? 1L : 0L);
         curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, hawkbit_config->ssl_verify ? 1L : 0L);
+
+        if (g_once_init_enter(&curl_share_initialized)) {
+                for (gint i = 0; i <= CURL_LOCK_DATA_LAST; i++)
```

Comment:
Maybe this deserves a short comment explaining that curl shopt requires a different lock for each data type (that is initialized here)?

Reply:
Done.